contracts/SpokePool.sol

```diff
@@ -147,19 +182,73 @@ abstract contract SpokePool is Testable, Lockable, MultiCaller {
     }
 
     function initiateRelay(
-        uint256 originChain,
-        address sender,
+        uint256 originChainId,
         uint256 amount,
-        address recipient,
-        uint256 relayerFee,
-        uint256 realizedLpFee
-    ) public {}
+        uint64 depositId,
+        uint64 relayerFeePct,
+        uint64 realizedLpFeePct,
+        address destinationToken,
+        address sender,
+        address recipient
+    ) public {
+        // We limit the relay fees to prevent the user spending all their funds on fees.
+        require(relayerFeePct <= 0.5e18 && realizedLpFeePct <= 0.5e18, "invalid fees");
+
+        // Use new relay count as unique ID for new relay.
+        relays.push(
+            RelayData(
+                recipient,
+                destinationToken,
+                relayerFeePct,
+                realizedLpFeePct,
+                amount, // total relay amount
+                0 // total amount filled
+            )
+        );
+        uint64 relayId = uint64(relays.length - 1);
```

Comment:
nit: you might be able to avoid the `- 1` if you move this above the push call.

Reply:
See above but I think I'm going to remove `relayId` in favor of a `relayHash` which is a hash of the relay data + deposit data